src/Compilers/VisualBasic/Portable/Lowering/AsyncRewriter/AsyncRewriter.vb

```diff
@@ -85,9 +85,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
             stateMachineType = New AsyncStateMachine(slotAllocatorOpt, compilationState, method, methodOrdinal, kind)
 
-            If compilationState.ModuleBuilderOpt IsNot Nothing Then
-                compilationState.ModuleBuilderOpt.CompilationState.SetStateMachineType(method, stateMachineType)
-            End If
+            Debug.Assert(compilationState.ModuleBuilderOpt IsNot Nothing)
```

Comment:
I think we can drop these assertion. ModuleBuilderOpt is now non-null during the entire lowering phase. Rather then assert so on every use add a doc comment on it saying so.